### PR TITLE
[tuner] return new checkpoint in result grid.

### LINKED ISSUE #23295: [ml] `Tuner.fit` does not return ML Checkpoint
Running `tuner.fit()[0].checkpoint` returns a `ray.tune.Checkpoint` object, not a `ray.ml.Checkpoint` object.

### LINKED ISSUE #23303: [ml] typo in tune docstring
https://github.com/ray-project/ray/blob/04ab27dcbf61226d5b53ef3f5375a98308d6a460/python/ray/tune/tuner.py#L60

We should fix up the docstring so that it actually runs

### PR DESCRIPTION
**Pull request by xwjiang2010**
## Related issue number

Closes #23295
Closes #23303

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by xwjiang2010** [comment]
@amogkam Consider the timeline, I prefer to go down this route to unblock merging the rest of PRs. wdyt?

**Comment by amogkam** [comment]
This sgtm. Should we add a test for this on the tuner side?

**Comment by xwjiang2010** [comment]
@amogkam Added some basic coverage. Will update it once settling on a long term plan with Kai.

**Review by amogkam — APPROVED** [review]
Lgtm, but I think you accidentally pushed over all the experiment result files

**Comment by amogkam** [comment]
just fyi this currently doesn't work if `trial.checkpoint.value` is None. I had to patch in `if trial.checkpoint.value else None` into #23282.

btw, doesn't seem like Tuner is being tested right now? Should we slot that into beta?

**Review by krfricke — CHANGES_REQUESTED** [review]
The actual fix looks good to me, but I'm against committing quasi-binary checkpoints/states into CI. If we ever change the checkpoint format we will need to re-generate these, there are no instructions on how to do so and it is a test maintenance overhead we don't want to have.

Instead, please either construct an ExperimentAnalysis object and populate the trials manually, or if really required, construct a trial runner, add trials + their results, call the `trial_runner.checkpoint()` method to a temporary directory and use that checkpoint.

**Review by krfricke — APPROVED** [review]
Looks good - is the actual change already merged (since this is just a test)?

File: python/ray/tune/tuner.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ class Tuner:
             },
         }
         tuner = Tuner(trainable=trainer, param_space=param_space,
-            run_config(name="my_tune_run"))
+            run_config(name="my_tune_run")
```
[the remark this answers is not in the document]
Comment (xwjiang2010) [reply python/ray/tune/tuner.py]: fixed the typo.

**Comment by xwjiang2010** [comment]
> Looks good - is the actual change already merged (since this is just a test)?

Yes that's correct :)

**Pull request merged**